modules/nextflow/src/main/groovy/nextflow/datasource/SraExplorer.groovy

```diff
@@ -264,23 +284,35 @@ class SraExplorer {
             missing << acc
             return null
         }
+        // modify this does not work //To do catch 500 http error
+        /*else if (result ==~  "A column id supplied is not valid")
+            log.debug "Not a valid SRA field =$fields"
+            missing << acc
+            return null*/
 
         return result
     }
-
-    protected getFastqUrl(String acc) {
-        def text = readRunFastqs(acc)
+    // to do here I dont need fieldssRA
+    protected getFastqUrl(String acc, fieldsSRA) {
+        def text = readRunFastqs(acc, fieldsSRA)
         if( !text )
             return
-
+        // modify
         def lines = text.trim().readLines()
-        def files = lines[1].split(';')
-        def result = new ArrayList(files.size())
+        def files = lines[1].split('\t')[0].split(';')
+        def fields =  lines[1].split('\t').drop(1)// TODO check whether fields exists
+        def result_files = new ArrayList(files.size())
         for( def str : files ) {
-            result.add( FileHelper.asPath("ftp://$str") )
+            result_files.add( FileHelper.asPath("ftp://$str") )
         }
+        def result_f = result_files.size()==1 ? result_files[0] : result_files
+        def result = new ArrayList(fields.size() + 1)
 
-        return result.size()==1 ? result[0] : result
+        result.add (result_f)
+        for( def it : fields ) {
+            result.add( it )
+        }
+        return result
```

Comment:
If this is implemented using a map then we have to declare which are the fields that can be retrieved from the query, there are many as you can see in this [example](https://www.ebi.ac.uk/ena/data/warehouse/filereport?result=read_run&accession=SRR307897). To avoid this, I will rather just use a list and a for, what do you think?

Reply:
The retrieve file has 2 lines 

```
study_accession	secondary_study_accession	sample_accession	secondary_sample_accession	experiment_accession	run_accession	submission_accession	tax_id	scientific_name	instrument_platform	instrument_model	library_name	nominal_length	library_layout	library_strategy	library_source	library_selection	read_count	base_count	center_name	first_public	last_updated	experiment_title	study_title	study_alias	experiment_alias	run_alias	fastq_bytes	fastq_md5	fastq_ftp	fastq_aspera	fastq_galaxy	submitted_bytes	submitted_md5	submitted_ftp	submitted_aspera	submitted_galaxy	submitted_format	sra_bytes	sra_md5	sra_ftp	sra_aspera	sra_galaxy	cram_index_ftp	cram_index_aspera	cram_index_galaxy	sample_alias	broker_name	sample_title	nominal_sdev	first_created
PRJNA30709	SRP007461	SAMN00634070	SRS214591	SRX082565	SRR307897	SRA039973	9606	Homo sapiens	ILLUMINA	Illumina Genome Analyzer II	GSM758559: CshlLong_RnaSeq_GM12878_cell_longPolyA		PAIRED	RNA-Seq	TRANSCRIPTOMIC	cDNA	32214376	4896585152	GEO	2011-07-28	2015-06-19	Illumina Genome Analyzer II sequencing; GSM758559: CshlLong_RnaSeq_GM12878_cell_longPolyA	GSE30567: Long RNA-seq from ENCODE/Cold Spring Harbor Lab	GSE30567	GSM758559: lab_RnaSeq_GM12878_cell_longPolyA	GSM758559_1	2392496788;2436856506	cdeb9d2450ca0bfc678968ac33f551a6;a1e9011d47d5b0462b9d50402a92dc6d	ftp.sra.ebi.ac.uk/vol1/fastq/SRR307/SRR307897/SRR307897_1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/fastq/SRR307/SRR307897/SRR307897_2.fastq.gz	fasp.sra.ebi.ac.uk:/vol1/fastq/SRR307/SRR307897/SRR307897_1.fastq.gz;fasp.sra.ebi.ac.uk:/vol1/fastq/SRR307/SRR307897/SRR307897_2.fastq.gz	ftp.sra.ebi.ac.uk/vol1/fastq/SRR307/SRR307897/SRR307897_1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/fastq/SRR307/SRR307897/SRR307897_2.fastq.gz							3499798854	2ee356e58e9765d6b26bde4b119b99be	ftp.sra.ebi.ac.uk/vol1/srr/SRR307/SRR307897	fasp.sra.ebi.ac.uk:/vol1/srr/SRR307/SRR307897	ftp.sra.ebi.ac.uk/vol1/srr/SRR307/SRR307897				GSM758559		CSHL_RnaSeq_GM12878_cell_longPolyA (superseded by GSE86658)		2011-07-28
```

1. Read the first line, 
2. tokenize by `\t` and keep the result list 
3. Read the second line 
4. tokenize by `\t`
5. iterate over the one of the two list and create a pair taking the i-st element from the first list as key and i-st element as the value from the second list